### ISSUE TITLE
Optionally download node+npm

### ISSUE THREAD

**Issue by mvysny** [vaadin/vaadin-gradle-plugin#22]
It would be great to use the https://plugins.gradle.org/plugin/com.github.node-gradle.node plugin (compatible also with Gradle 6) to optionally download the node distro in a way compatible with Vaadin. We could even provide a task for that.

The necessary code is present in given patch file 
[use_local_node_installation.patch.txt](https://github.com/vaadin/vaadin-gradle-plugin/files/4138625/use_local_node_installation.patch.txt)

**Comment by mvysny** [comment]
Actually maybe I can get away simply by hacking into FrontendToolsLocator.frontendToolsLocator? I'll investigate.

**Comment by mvysny** [comment]
The following task has been added: `vaadinPrepareNode` - it will prepare a local distribution of node.js and npm for use by Vaadin. However, it will NOT download the node.js itself.

In order to do that, you will need the [com.github.node-gradle.node](https://plugins.gradle.org/plugin/com.github.node-gradle.node) plugin.
Add the following to your `build.gradle` file:

```groovy
plugins {
    id "com.github.node-gradle.node" version "2.2.1"  // in order for vaadinPrepareNode to work
}
node {
    version = "10.15.2"
    download = true
    // to download node+npm, just run the `vaadinPrepareNode` task
}
```

In your development environment, all you need to do is to run:
```bash
./gradlew vaadinPrepareNode
```
to download and prepare a local distribution of node.js. You only need to run this once:
a folder named `node/` is created in the project directory, which will be used by Vaadin
from now on.

In your CI, don't forget to call the `vaadinPrepareNode` before the `vaadinPrepareFrontend` task:
```bash
./gradlew clean vaadinPrepareNode vaadinBuildFrontend build
```

**Comment by mvysny** [comment]
Implemented in 0.3.0
